src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,118 @@
+package duke.command;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import duke.command.TaskList;
+import duke.command.Ui;
+import duke.command.Storage;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.DukeException;
+
+/**
+ * Represents a Command manager that adds new Task to TaskList, deletes
+ * tasks and printing the list of tasks.
+ * Contains the fields TaskList that contains the list of Task,
+ * Ui that prints the message for each commands and Storage that
+ * updates the files when the list of Task is modified.
+ */
+public class Parser {
+
+    private TaskList taskList;
+    private Ui ui;
+    private Storage storage;
+
+    /**
+     * Constructs parser object that takes in TaskList,
+     * Ui and Storage as parameter.
+     *
+     * @param taskList  list of tasks contained in TaskList object.
+     * @param ui User Interface object responsible for printing messages.
+     * @param storage Storage that loads and updates files when tasks are
+     *                deleted or added to the list.
+     */
+    public Parser(TaskList taskList, Ui ui, Storage storage) {
+        this.taskList = taskList;
+        this.ui = ui;
+        this.storage = storage;
+    }
+
+    /**
+     * Parse a one line input into a Task or Command.
+     * Print message is produced every time an input is passed in.
+     *
+     * @param input  Command to be executed or Task to be created.
+     * @throws DukeException  if Task has incomplete description.
+     */
+    public void parseLineInput(String input) throws DukeException, IOException, ParseException {
+        String firstWord = input.split(" ")[0];
+        //ArrayList<Task> list = taskList.getList();
+        if (firstWord.equals("bye")) {
```

Comment:
Your logic is smooth and easy to understand, well done! However, do you think it would be tidier and more readable to use switch statements instead of a large number of switch statements?